### ISSUE TITLE
[Bug] Bad results clone voice when loading with model_path and config_path

### ISSUE THREAD

**Issue by OlegRuban-ai** [idiap/coqui-ai-TTS#134]
### Describe the bug

I used 2 options to download model:

1. tts = TTS(
    model_path="/XTTS",
    config_path="/XTTS/config.json",
    progress_bar=True,
).to('cuda')

2. tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2", progress_bar=True, gpu=True

In the first option, the result after generation is much worse than in the second with standard loading. Why? How to load all configuration files correctly and from where?

I took the model for model_path and the config from here: https://huggingface.co/coqui/XTTS-v2/tree/main

### To Reproduce

1. tts = TTS(
    model_path="/XTTS",
    config_path="/XTTS/config.json",
    progress_bar=True,
).to('cuda')

2. tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2", progress_bar=True, gpu=True

### Environment

```shell
TTS 0.24.3
PyTorch 1.8
```

**Comment by eginhard** [comment]
Can you fix the seed as below and compare both? For me both methods of loading result in the exact same output:

```python
import os

import torch
from trainer.io import get_user_data_dir

from TTS.api import TTS

model_name = "tts_models/multilingual/multi-dataset/xtts_v2"

xtts1 = TTS(model_name).to("cuda")

xtts_dir = os.path.join(get_user_data_dir("tts"), "--".join(model_name.split("/")))
xtts2 = TTS(model_path=xtts_dir, config_path=os.path.join(xtts_dir, "config.json")).to("cuda")

torch.manual_seed(123)
out1 = xtts1.tts("This is a test", speaker="Ana Florence", language="en")

torch.manual_seed(123)
out2 = xtts2.tts("This is a test", speaker="Ana Florence", language="en")

assert out1 == out2
```

**Comment by OlegRuban-ai** [comment]
@eginhard 

Thank you, I managed to find an error in my code. But there is still a problem with the limit of 182 tokens for the Russian language. To fix this when loading from HF, I used a replacement in tokenizer.py, but when loading from a local file, that path is no longer used for some reason (tts/layers/xtts/tokenizer.py). How can I bypass the restriction?

**Comment by eginhard** [comment]
The results are also the same when using a `speaker_wav`, but again you need to correctly fix the seed:
```python
import os

import torch
from trainer.io import get_user_data_dir

from TTS.api import TTS

model_name = "tts_models/multilingual/multi-dataset/xtts_v2"

xtts1 = TTS(model_name).to("cuda")

xtts_dir = os.path.join(get_user_data_dir("tts"), "--".join(model_name.split("/")))
xtts2 = TTS(model_path=xtts_dir, config_path=os.path.join(xtts_dir, "config.json")).to("cuda")

wav_fn = "tests/data/ljspeech/wavs/LJ001-0032.wav"

torch.manual_seed(123)
out1 = xtts1.tts("This is a test", speaker_wav=wav_fn, language="en")

torch.manual_seed(123)
out2 = xtts2.tts("This is a test", speaker_wav=wav_fn, language="en")

assert out1 == out2
```

You can increase the token limits in the code, but the results will be worse because the model hasn't been trained on such long sequences. Coqui automatically splits the input into sentences when using it like I shared.
